react/features/base/util/uri.js

```diff
@@ -136,6 +136,22 @@ export function getLocationContextRoot({ pathname }: { pathname: string }) {
             : pathname.substring(0, contextRootEndIndex + 1));
 }
 
+/**
+ * Normalizes the room name of an URI allowing only a predefined set
+ * of characters. returns undefined if the room name has a falsy value.
+ *
+ * @private
+ * @param {string} uriRoomSegment - The room segment of the URI.
+ * @returns {?string}
+ */
+function _normalizeRoomName(uriRoomSegment: string) {
+    if (!uriRoomSegment) {
+        return undefined;
+    }
+
+    return uriRoomSegment.replace(/[^A-Za-z0-9_\\-\\.]/g, '');
```

Comment:
Like you mentioned, I'm not sure this covers all the cases. For example, % could be used for URI encoding but that could get parsed, no? Is the URL created by this regex actually accurate for the meeting?

Reply:
@zbettenbuk and I agreed that he'll look at the patterns used by meet.jit.si, beta.meet.jit.si, and the other production services that we're running which potentially have different patterns and try to identify at least common special characters which don't seem to be accepted anywhere.

Reply:
What Lenny says. There's https://tools.ietf.org/html/rfc3986#section-2 which talks about what goes into a URL. On top of that we obviously have some constraints of our own imposed on our servers.

Reply:
We agreed with Aaron that based on the RFC excluding these characters should be a good start:

gen-delims  = ":" / "/" / "?" / "#" / "[" / "]" / "@"
sub-delims  = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="

Reply:
I skipped the "/" character as Aaron mentioned that that is used to delimit group/room.

However testing the feature seems like there are characters that cannot be excluded as they're causing disturbance in the force BEFORE the room name detection. E.g. if you enter anything in the field starting with a "?" or "#" - I guess they are parsed as query and hash segments. Maybe the URI parsing mechanism could use a review later. Or maybe that's expected behaviour and then you can ignore this comment.